extract_blocks.py

```diff
@@ -9,9 +10,9 @@
     os.mkdir(os.path.split(sys.argv[1])[1] + ".extract")
     os.chdir(os.path.split(sys.argv[1])[1] + ".extract")
     with open("superblock.bin", "w+b") as f:
-        f.write(fs.read(FATX.SUPERBLOCK_SIZE))
+        f.write(fs.read(SuperBlock.SUPERBLOCK_SIZE))
     with open("fat.bin", "w+b") as f:
-        fat_size = FATX.Filesystem._calc_fat_size(size)
+        fat_size = FATX.Filesystem._calc_fat_size(size, 512*32)
```

Comment:
Is 512 the sector size?
Isn't 32 the number of sectors per cluster?

Reply:
Yes and yes. I see where you are going here - `extract_blocks.py` will hopefully be removed from the project soonish. Its only purpose is to provide easy accessible binary data for debugging purposes that are much much smaller then the original image and also hopefully do not contain copyrighted material. This allows me to ask users of FATX-on-a-snake to run this script and send me over the extracted files, so I can take a look. However, this is good enough for _now_ but it will be unnecessary in the future, hence I want to remove it from the project. In order to do so I still have to gain more confidence that FATX-on-a-snake works good enough and is somewhat error resistant so it will be able to produce text-based, human-readable log output. That log can then be send over for remote debugging and will be guaranteed be free from copyright material.

My reasoning behind picking `512*32` instead of automatically calculating it is, that when this script is run, something is most-likely heavily broken. i.e. if a user attempts to open the xbox hdd from offset `0x00` it won't see the `FATX` magic bytes, yet, `extract_blocks.py` will happily run and extract the "should-be superblock".